crates/router/src/core/payments/access_token.rs

```diff
@@ -196,3 +271,150 @@ pub async fn refresh_connector_auth(
     );
     Ok(access_token_router_data.response)
 }
+
+fn construct_access_token_router_data(
+    tracking_data: &ProcessTrackerAccessTokenData,
+    merchant_connector_account: &storage::MerchantConnectorAccount,
+) -> RouterResult<
+    types::RouterData<
+        api_types::AccessTokenAuth,
+        types::AccessTokenRequestData,
+        types::AccessToken,
+    >,
+> {
+    let connector_auth_type: types::ConnectorAuthType = merchant_connector_account
+        .connector_account_details
+        .clone()
+        .parse_value("ConnectorAuthType")
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "connector_account_details",
+        })?;
+
+    let access_token_request =
+        types::AccessTokenRequestData::try_from(connector_auth_type.clone())?;
+
+    Ok(types::RouterData {
+        flow: std::marker::PhantomData,
+        merchant_id: tracking_data.merchant_id.clone(),
+        connector: tracking_data.connector.clone(),
+        payment_id: tracking_data.payment_id.clone(),
+        attempt_id: tracking_data.attempt_id.clone(),
+        status: storage_models::enums::AttemptStatus::Pending,
+        payment_method: tracking_data.payment_method,
+        connector_auth_type,
+        description: None,
+        return_url: None,
+        address: types::PaymentAddress::default(),
+        auth_type: storage_models::enums::AuthenticationType::default(),
+        connector_meta_data: None,
+        amount_captured: None,
+        access_token: None,
+        session_token: None,
+        reference_id: None,
+        request: access_token_request,
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+    })
+}
+
+pub async fn get_access_token_retry_schedule_time(
+    db: &dyn db::StorageInterface,
+    connector: &str,
+    merchant_id: &str,
+    retry_count: i32,
+) -> Result<Option<time::PrimitiveDateTime>, errors::ProcessTrackerError> {
+    let pt_mapping_key = &format!("pt_mapping_refund_sync_{connector}");
+    let custom_merchant_mapping = db.find_config_by_key(pt_mapping_key).await?;
+
+    let connector_mapping = custom_merchant_mapping
+        .config
+        .parse_struct::<ConnectorPTMapping>("CustomPTMapping");
+
+    let mapping = match connector_mapping {
+        Ok(x) => x,
+        Err(err) => {
+            crate::logger::error!("Error: while getting connector mapping: {}", err);
```

Comment:
remove absolute path in function calls
```suggestion
            logger::error!("Error: while getting connector mapping: {}", err);
```